api/types/worker.proto

```diff
@@ -2,17 +2,14 @@ syntax = "proto3";
 
 package moby.buildkit.v1.types;
 
-import "github.com/gogo/protobuf/gogoproto/gogo.proto";
 import "github.com/moby/buildkit/solver/pb/ops.proto";
 
-option (gogoproto.sizer_all) = true;
-option (gogoproto.marshaler_all) = true;
-option (gogoproto.unmarshaler_all) = true;
+option go_package = "github.com/moby/buildkit/api/types;moby_buildkit_v1_types";
 
 message WorkerRecord {
 	string ID = 1;
 	map<string, string> Labels = 2;
-	repeated pb.Platform platforms = 3 [(gogoproto.nullable) = false];
+	repeated pb.Platform platforms = 3;
```

Comment:
Could we get some backward compatibility proving unit tests for the structs where the proto definition changed? Tests that use binary assets generated from previous version.

Reply:
Let me check. I could write a few, but probably not for all of them. I feel that it is more like testing protos, not Buildkit.

Reply:
It may make sense to do that in master first since this PR changes protos :)

Reply:
Opened #4570.